README.md

```diff
@@ -1,289 +1,590 @@
+    ```plain
+    service.binding”:
+      "path={.status.data.dbConfiguration},objectType=ConfigMap”
+    ```
+1.  Mount an entry from a `ConfigMap` into the binding `Secret`
+    ```plain
+    “service.binding/certificate”:
+      "path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=certificate"
+    ```
+1.  Mount an entry from a `ConfigMap` into the binding `Secret` with a different key
+    ```plain
+    “service.binding/timeout”:
+      “path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=db_timeout”
+    ```
+1.  Mount a resource defintion value into the binding `Secret`
+    ```plain
+    “service.binding/uri”:
+      "path={.status.data.url}"
+    ```
+1.  Mount a resource definition value into the binding `Secret` with a different key
+    ```plain
+    “service.binding/uri":
+      "path={.status.data.connectionURL}”
+    ```
+1.  Mount the entries of a collection into the binding `Secret` selecting the key and value from each entry
+    ```plain
+    “service.binding/endpoints”:
+      "path={.status.bootstrap},elementType=sliceOfMaps,sourceKey=type,sourceValue=url"
+    ```
+
+## Mapping Existing Values to New Values
+
+Many applications will not be able to consume the secrets exposed by Provisioned Services directly.  Teams creating Provisioned Services do not know how their services will be consumed, teams creating Applications will not know what services will be provided to them, different language families have different idioms for naming and style, and more.  Users should have a way of describing a mapping from existing values to customize the provided entries to ones that are usable directly by their applications.  This specification is described as an extension to the [Service Binding](#service-binding) specification and assumes full compatibility with it.
+
+A Service Binding Resource **MAY** define a `.spec.mappings` which is an array of `Mapping` objects.  A `Mapping` object **MUST** define `name` and `value` entries.  The value of a `Mapping` **MAY** contain zero or more tokens beginning with `((`, ending with `))`, and encapsulating a binding `Secret` key name.  The value of this `Secret` entry **MUST** be substituted into the original `value` string, replacing the token.  Once all tokens have been substituted, the new `value` **MUST** be added to the `Secret` exposed to the resource represented by `application`.
+
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name:         # string
+spec:
+  name:         # string, optional, default: .metadata.name
+
+  application:  # PodSpec-able resource ObjectReference
+    apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  service:      # Provisioned Service-able resource ObjectReference
+    apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  mapping:      # []Mapping, optional
```

Comment:
@arthurdm is this the one we've been referring to `dataMapping` so far ? I'm good!

Reply:
yup.  and I believe it should be plural (`mappings`), @nebhale?   As we can construct more than 1 combined mapping.

Reply:
Yep, that's the one.  Syntax is slightly different, but an arbitrary substitution of as many values into and out of the `Secret`.